src/main/java/org/reaktivity/nukleus/http/internal/util/HttpUtil.java

```diff
@@ -35,6 +51,37 @@ public static void appendHeader(
         payload.append(initCapsName).append(": ").append(value).append("\r\n");
     }
 
+    public static boolean isValidPath(
+        DirectBuffer path)
+    {
+        boolean isPathValid = true;
+        for (int i = 0; i < path.capacity(); i++)
+        {
+            byte charByte = path.getByte(i);
+
+            if (((charByte & 0x20) == 0) ||
+                ((charByte & 0x80) == 0x80) ||
+                (charByte == ASCII_32) ||
+                (charByte == ASCII_34) ||
+                (charByte == ASCII_37) ||
+                (charByte == ASCII_60) ||
+                (charByte == ASCII_62) ||
+                (charByte == ASCII_92) ||
+                (charByte == ASCII_94) ||
+                (charByte == ASCII_96) ||
+                (charByte == ASCII_123) ||
+                (charByte == ASCII_124) ||
+                (charByte == ASCII_125) ||
+                (charByte == ASCII_127))
+            {
+                isPathValid = false;
+                break;
+            }
+        }
+        return isPathValid;
+    }
```

Comment:
We can still get the benefit of striding by `long` for the low range and high range checks which use masks.
```
public static boolean isPathValid(
    DirectBuffer path)
{
    boolean valid = true;
    int capacity = path.capacity();

    long_loop:
    for (int index = 0; capacity > Long.BYTES; index += Long.BYTES, capacity -= Long.BYTES)
    {
        long candidate = path.getLong(index);

        if ((candidate & ASCII_LOW_RANGE_LONG_MASK) == 0L) || (candidate & ASCII_HIGH_RANGE_LONG_MASK) != 0L)
        {
            valid = false;
            break long_loop;
        }

        while (candidate != 0L)
        {
            switch (((int)(candidate & 0x0000_0000_0000_00FFL))
            {
            case ASCII_SPACE:
            case ASCII_DOUBLE_QUOTES:
            case ASCII_PERCENT_SIGN:
            case ASCII_LESS_THAN:
            case ASCII_GREATER_THAN:
            case ASCII_BACKSLASH:
            case ASCII_CARET:
            case ASCII_GRAVE:
            case ASCII_OPEN_BRACE:
            case ASCII_VERTICAL_BAR:
            case ASCII_CLOSE_BRACE:
            case ASCII_DELETE:
                valid = false;
                break long_loop;
            default:
                candidate >>= 1;
                break;
            }
        }
    }

    if (valid)
    {
        byte_loop:
        for (int index = capacity; capacity > 0; index++, capacity--)
        {
            byte candidate = path.getByte(index);

            if ((candidate & ASCII_LOW_RANGE_MASK) == 0) || (candidate & ASCII_HIGH_RANGE_MASK) != 0)
            {
                valid = false;
                break byte_loop;
            }

            switch (candidate)
            {
            case ASCII_SPACE:
            case ASCII_DOUBLE_QUOTES:
            case ASCII_PERCENT_SIGN:
            case ASCII_LESS_THAN:
            case ASCII_GREATER_THAN:
            case ASCII_BACKSLASH:
            case ASCII_CARET:
            case ASCII_GRAVE:
            case ASCII_OPEN_BRACE:
            case ASCII_VERTICAL_BAR:
            case ASCII_CLOSE_BRACE:
            case ASCII_DELETE:
                valid = false;
                break byte_loop;
            default:
                break;
            }
        }
    }

    return valid;
}
```